dropwizard-core/src/main/java/io/dropwizard/server/AdminServletFactory.java

```diff
@@ -0,0 +1,235 @@
+ *         <td>Description</td>
+ *     </tr>
+ *     <tr>
+ *         <td>metricsEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether metrics information will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>metricsUri</td>
+ *         <td>/metrics</td>
+ *         <td>The relative URI for metrics retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>pingEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether a {@code ping} liveness function will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>pingUri</td>
+ *         <td>/ping</td>
+ *         <td>The relative URI for liveness response retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>threadsEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether thread dump information will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>threadsUri</td>
+ *         <td>/threads</td>
+ *         <td>The relative URI for thread dump retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>healthcheckEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether health check information will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>healthcheckUri</td>
+ *         <td>/healthcheck</td>
+ *         <td>The relative URI for health check retrieval.</td>
+ *     </tr>
+ *     <tr>
+ *         <td>cpuProfileEnabled</td>
+ *         <td>true</td>
+ *         <td>
+ *             Determines whether CPU profiling results will be available via the
+ *             {@link com.codahale.metrics.servlets.AdminServlet AdminServlet} endpoint.
+ *         </td>
+ *     </tr>
+ *     <tr>
+ *         <td>cpuProfileUri</td>
+ *         <td>/pprof</td>
+ *         <td>The relative URI for CPU profiling retrieval.</td>
+ *     </tr>
+ * </table>
+ *
+ * @since 2.0
+ */
+public class AdminServletFactory {
+
+    private boolean metricsEnabled = true;
+
+    private String metricsUri = AdminServlet.DEFAULT_METRICS_URI;
+
+    private boolean pingEnabled = true;
+
+    private String pingUri = AdminServlet.DEFAULT_PING_URI;
+
+    private boolean threadsEnabled = true;
+
+    private String threadsUri = AdminServlet.DEFAULT_THREADS_URI;
+
+    private boolean healthcheckEnabled = true;
+
+    private String healthcheckUri = AdminServlet.DEFAULT_HEALTHCHECK_URI;
+
+    private boolean cpuProfileEnabled = true;
+
+    private String cpuProfileUri = AdminServlet.DEFAULT_CPU_PROFILE_URI;
```

Comment:
Do you think it might make sense to model this as a map of string -> (boolean, URI)?

That way people could add their own keys and enable certain functionality on their admin connector.

```yaml
adminServlets:
  metrics:
    enabled: true
    uri: /metrics
  ping:
    enabled: true
    uri: /ping
  threads:
    enabled: false
```

Reply:
This is interesting; it definitely groups the configuration more clearly. Is it still reasonable that anything not explicitly configured is enabled and available at the default endpoint?

Could you please let me know if there's an example of this type of functionality somewhere else I could use as a model? It seems like creating a simple object that has a `boolean` and `String` properties would be sufficient. I'd also like to model the default value pattern (if they don't provide a URL, for example) on existing code if there's any you could point me to.

Thanks!

Reply:
@joschi,

When you say a user would "add their own keys", does that mean you want to be able to have sparse configuration? That's what your example looks like. Also, the `AdminServlet` configuration won't support ad hoc keys, so it seems that's not where you want to go.

Could you please help me understand if I'm missing anything?

Reply:
@joschi,

I've pushed a potential update that implements the pattern as I understand it. If this works for you, I'll update the documentation (class and RST), and push another update.

Thanks!

Reply:
@joschi,

I will address the conflicts as soon as I can; could you please review the direction so far to see if this is how you'd like the configuration to be structured?

Reply:
@joschi,

Could you please provide feedback on the direction so far? I can definitely pick this back up once I know if I'm going in the right direction.

Reply:
Hi @mschechter-bellese. So far, it looks good to me.

> Also, the `AdminServlet` configuration won't support ad hoc keys, so it seems that's not where you want to go.

This makes sense to me. I'd suggest to add a check, if there are any keys we cannot handle.